libevmcore/Assembly.cpp

```diff
@@ -171,61 +173,96 @@ ostream& dev::eth::operator<<(ostream& _out, AssemblyItemsConstRef _i)
 	return _out;
 }
 
-ostream& Assembly::streamRLP(ostream& _out, string const& _prefix) const
+string Assembly::getLocationFromSources(StringMap const& _sourceCodes, SourceLocation const& _location) const
+{
+	if (_location.isEmpty() || _sourceCodes.empty())
+		return "";
+
+	string toReturn = _sourceCodes.find(*_location.sourceName)->second;
+
+	if (_location.end < (int)toReturn.size() - 1)
+			return "";
+
+	toReturn = toReturn.substr(_location.start, _location.end);
+	auto newLinePos = toReturn.find_first_of("\n");
+	if (newLinePos != string::npos && newLinePos != toReturn.size() - 1)
```

Comment:
Please remove the second check, the string should not contain a newline even if it is the last character.